dto/apikey_dto.go

```diff
@@ -14,8 +13,24 @@ func AdaptApiKeyDto(apiKey models.ApiKey) ApiKey {
 	return ApiKey{
 		Id:             apiKey.Id,
 		OrganizationId: apiKey.OrganizationId,
-		Key:            apiKey.Key,
 		Description:    apiKey.Description,
 		Role:           apiKey.Role.String(),
 	}
 }
+
+type CreatedApiKey struct {
+	ApiKey
+	Key string `json:"key"`
+}
+
+func AdaptCreatedApiKeyDto(apiKey models.CreatedApiKey) CreatedApiKey {
+	return CreatedApiKey{
+		ApiKey: AdaptApiKeyDto(apiKey.ApiKey),
+		Key:    apiKey.Value,
+	}
+}
```

Comment:
Totally agree with you. I found it weird first, but I think this is due to the lack of "spread" operator in Go (like `...obj` in JS/TS).

In the end, using this kind of composition only for DTO, with the go nesting wrapped inside adapters should not be a problem IMO, but I do not have strong opinion between this, or duplicate all the code every time (since we do have CoPilot to do it for us and this should be written once).

(small) Bonus: having duplication for all those types makes them trully independent (so easier to introduce independant changes on each endpoints).

This is just a lazy way to do things

Reply:
yea agreed, let's go for this solution 👍